### DISCUSSION TITLE
Multi tenant advice

### DISCUSSION THREAD

**Discussion by rgustavsson** [linq2db/linq2db#4590] [General]
Hej!

I'm looking for some advice.

We have a multi tenant platform which consists of a server application, .NET using Linq2DB, and a postgresql database. 
All tenant data is stored in the very same database.
So we use the same connection string for all tenants since it's just a single database.

We now have a scenario where one of the tenants need to host their own database and so we were thinking of creating the connection string during runtime but after many minutes of reading different forums we think that this may not be supported by linq2db, that our approach doesn't fit the performance aspect of what linq2db is supposed to deliver.

Our tenants/databse would still have the very same database schema and we've seen that there is a "ChangeDatabaseName" method available to use. Would that be a way forward perhaps? That we, in our database context constructor, change the database name, and this would only affect the instance of the context that was initialized via DI? 

Any other advice?

**Comment by MaceWindu** [comment]
Just add new configuration at runtime with any unique name and use it for customer-specific connections
```cs
DataConnection.AddConfiguration(connectionName, connectionString, provider);
```

**Comment by rgustavsson** [comment]
Ok, so we can e.g. have 100 different configurations that all point to different database servers (different locations/IPs) ?

**Reply by MaceWindu** [reply]
Yes, any sane number or dynamic/static configurations to any supported datase types in single process

**Reply by rgustavsson** [reply]
That sounds great.

Many thanks for taking your time.
